src/main/java/com/arpnetworking/metrics/impl/TsdMetricsFactory.java

```diff
@@ -107,37 +205,66 @@ protected TsdMetricsFactory(final Builder builder) {
     public static class Builder {
 
         /**
+         * Public constructor.
+         */
+        public Builder() {
```

Comment:
I know there is no absolute answer for this topic, so I am not religious towards my point.

I'd like to just record some quick points (sorry if they may be redundant from previous conversation :-) )

- Telescoping constructor typically happens when there is a mix of required and optional params. If all parameters are required you end up with a single constructor and you may even consider using a static factory in that case vs a Builder() (Though i think that would loose another benefit of the builder, which is naming the parameters)

- I think we are in general agreement that compile time pattern enforcement (if possible) would be way nicer than discovering issue in runtime

- Regarding that  that you test your code, this is a very valid point, but please notice that the failure here is reported using a warning in the log file, so unless the clients writes test cases to validate that the metrics factory was constructed as specified, it might be easy to slip this issue

- Given the above point, I wonder if there was a consideration to throw an exception in the Metrics Factory Building rather than log a warning. Unlike metrics objects, I would assume the metrics factory is expected to have few constructions during the startup of the app, so a fail fast would easily  ring bells in many test cases

- Last thing, more of an FYI, there is another option to have the best of both worlds, but it might not play nice with serialization, and I personally prefer it only when there are few required parameters:    http://www.jayway.com/2012/02/07/builder-pattern-with-a-twist/ 
For reference purpose, I gave it the name of Mandatory Builder Pattern, since I didn't find an official distinct name for it :smile: